projects/packages/my-jetpack/_inc/components/product-cards-section/index.tsx

```diff
@@ -107,67 +106,18 @@ interface ProductCardsSectionProps {
 }
 
 const ProductCardsSection: FC< ProductCardsSectionProps > = ( { noticeMessage } ) => {
-	const {
-		data: { ownedProducts },
-		isLoading,
-	} = useProductsByOwnership();
-
-	const [ isLoadingProducts, setIsLoadingProducts ] = useState( true );
-
-	useEffect( () => {
-		if ( isLoading ) {
-			return;
-		}
-
-		// This adds a slight delay to the loading status change to prevent
-		// a brief moment in time where the section was not visible at all
-		// between the isLoading = true and isLoading = false states.
-		// This issue was causing a flicker effect.
-		requestAnimationFrame( () => setIsLoadingProducts( false ) );
-	} );
-
-	const { canUserViewStats } = getMyJetpackWindowInitialState();
-
-	const filterProducts = useCallback(
-		( products: JetpackModule[] ) => {
-			const productsWithNoCard = [
-				'extras',
-				'scan',
-				'security',
-				'ai',
-				'creator',
-				'growth',
-				'complete',
-				'site-accelerator',
-				'newsletter',
-				'related-posts',
-				'brute-force',
-			];
-
-			// If the user cannot view stats, filter out the stats card
-			if ( ! canUserViewStats ) {
-				productsWithNoCard.push( 'stats' );
-			}
-
-			return products.filter( product => {
-				return ! productsWithNoCard.includes( product );
-			} );
-		},
-		[ canUserViewStats ]
-	);
-
-	const filteredOwnedProducts = filterProducts( ownedProducts );
+	const { filteredOwnedProducts, isLoading } = useFilteredProducts();
```

Comment:
FYI @manzoorwanijk I extracted this into a hook to avoid duplicated code after the split.

Reply:
Thanks